File: api/app/models/loan.rb
class Loan < ApplicationRecord
  validates :loan_officer_name, :property_address, :amount, presence: true
  validates :property_address, uniqueness: true

  has_and_belongs_to_many :borrowers

  def as_json(options = {})
    super(options).merge({ borrowerCount: borrowers.count })
  end
end

File: client/src/types/loan.ts
import Base from "./base";

export class Loan extends Base {
  constructor(
    loanOfficerName: string,
    propertyAddress: string,
    amount: number,
    borrowerCount: number,
    borrowerIds?: number[]
  ) {
    super();

    this.loanOfficerName = loanOfficerName;
    this.propertyAddress = propertyAddress;
    this.amount = amount;
    this.borrowerCount = borrowerCount;
    this.borrowerIds = borrowerIds;
  }

  loanOfficerName: string;
  propertyAddress: string;
  amount: number;
  borrowerCount: number;
  borrowerIds?: number[];
}
